client/allocrunner/tasklifecycle/testing.go

```diff
@@ -0,0 +1,56 @@
+package tasklifecycle
+
+import (
+	"testing"
```

Comment:
We do not want to pull the `testing` package into our core binary. Use https://github.com/mitchellh/go-testing-interface and its `testing.TB` instead of `testing` and `*testing.T`

Reply:
in theory we should be able to set  `//go:build test` and make sure of that 

haven't tried it though :sweat_smile:

Reply:
Wait did that get added? I thought it had been rejected?! https://github.com/golang/go/issues/21360

Reply:
Oh wow, I totally just assumed that would work :man_facepalming:

Reply:
I think we all have. Kind of boggles my mind that even Rob Pike asked for it and they're like `nah`. Test helpers right inside the package they're intended to be used with seems like a cool idea to me. :shrug:

Reply:
Oh big TIL. Smells like a new semgrep rule 😄

Reply:
The [version we're currently using](https://github.com/hashicorp/nomad/blob/v1.3.3/go.mod#L95) doesn't have the [`TB` interface](https://github.com/mitchellh/go-testing-interface/blob/v1.14.1/testing.go), so I'm going to use just `T`.